src/smvlang/smv_parse_tree.h

```diff
@@ -118,100 +118,110 @@ class smv_parse_treet
     typedef std::list<itemt> item_listt;
     item_listt items;
 
-    void add_item(
-      itemt::item_typet item_type,
-      const exprt &expr,
-      const source_locationt &location)
+    void
+    add_item(itemt::item_typet item_type, exprt expr, source_locationt location)
     {
-      items.push_back(itemt());
+      items.push_back(itemt{});
       items.back().item_type=item_type;
-      items.back().expr=expr;
-      items.back().location=location;
+      items.back().expr = std::move(expr);
+      items.back().location = std::move(location);
```

Comment:
Perhaps `itemt` should have a proper constructor, at which point it should be possible to use
`items.emplace_back(item_type, std::move(expr), std::move(location));`. (Which in turns makes me wonder whether we could then just get rid of `add_item` as a wrapper altogether.)